cmds/readlink/readlink.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2013-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// readlink display value of symbolic link file
+//
+// Synopsis:
+//     readlink [OPTIONS] FILE
+//
+// Options:
+//     -f: follow
+//     -v: verbose
+//
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+var (
+	follow  = flag.Bool("f", false, "follow recursively")
+	verbose = flag.Bool("v", false, "report error messages")
+)
+
+func isLink(file string) error {
+	var err error
```

Comment:
no need to declare err here, the line below already derives its type.

Reply:
Actually, the easiest way to see if a link is a link is readlink and look for an error.
So just do the readlink below, and see if the error is "this is not a symlink". If it is not, then
return that error. If *follow is false, you're done -- you got the value in Readlink.

doing a stat and a readlink is actually racy, and believe it or not, I've had it happen.